### PR TITLE
Added Pmqa Testsuite test in avocado [V2]

### PR DESCRIPTION
**Pull request by PraveenPenguin**
v1: #59 
1-inject all test through yaml
2-use os.path.join to combine dir
3-change style
4-change helper function return fail check status

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
job log : https://paste.fedoraproject.org/436066/74904465/

File: cpu/pmqa.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            ext_opt = 'hotplug_allow_cpu0=0'
+
+        cmd = '%s %s run_tests' % (testcase, ext_opt)
+
+        process.system('make -C %s' % cmd, ignore_status=True, shell=True)
+
+        result = process.run("grep -wF 'fail' %s" % log, ignore_status=True)
+
+        if not result.stdout == '':
+            self.log.info(result.stdout)
```
Comment (apahim) [inline cpu/pmqa.py]: Why to show an empty stdout?

File: cpu/pmqa.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    cpuidlee, thermal
+    """
+
+    def setUp(self):
+        '''
+        Build Pmqa Test
+        Source:
+        git://git.linaro.org/power/pm-qa.git
+        '''
+        self.failtest = 0
```
Comment (apahim) [inline cpu/pmqa.py]: Leftover?

File: cpu/pmqa.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            return True
+
+        return False
+
+    def test(self):
+        error = False
+
+        self.test_type = self.params.get('run_arg', default='cpufreq')
+
+        error = self._run_one(self.test_type)
```
Comment (apahim) [inline cpu/pmqa.py]: No need to have this `error`.
```
if not self._run(self.test_type)
    self.fail(...)
```

File: cpu/pmqa.py (hunk 1 above)
Comment (apahim) [inline cpu/pmqa.py]: Ops, nevermind. I mission d the `not` here. /me--

File: cpu/pmqa.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        if error:
+            self.fail("few tests failed")
```
Comment (apahim) [inline cpu/pmqa.py]: This message is misleading.

File: cpu/pmqa.py (hunk 3 above)
Comment (PraveenPenguin) [inline cpu/pmqa.py]: @apahim sure will do

File: cpu/pmqa.py (hunk 4 above)
Comment (PraveenPenguin) [inline cpu/pmqa.py]: sure

**Comment by PraveenPenguin** [comment]
@apahim raised New PR #65  with suggested change Please have a look .

Regards
Praveen

**Pull request closed without merging**